modules/mux/targets/host/source/kernel.cpp

```diff
@@ -259,13 +259,29 @@ mux_result_t hostQueryLocalSizeForSubGroupCount(mux_kernel_t kernel,
                                                 size_t *local_size_x,
                                                 size_t *local_size_y,
                                                 size_t *local_size_z) {
-  (void)kernel;
-  // FIXME: For a single sub-group, we know we can satisfy that with a
-  // work-group of 1,1,1. For any other sub-group count, we should ensure that
-  // the work-group size we report comes back through getKernelVariantForWGSize
-  // when it comes to run it. See CA-4784.
-  if (sub_group_count == 1) {
-    *local_size_x = 1;
+  host::kernel_variant_s variant;
```

Comment:
Is there any possibility of using some common code with the jit path here?

Reply:
In theory yes, in practice I think it would require a much bigger redesign. Right now, we don't even agree on the API: this function returns through pointers, whereas [`HostKernel::queryLocalSizeForSubGroupCount`](https://github.com/codeplaysoftware/oneapi-construction-kit/blob/cca28347df75b15572bf071000ceb1431621a86c/modules/compiler/targets/host/source/kernel.cpp#L182-L216) returns by value.